src/main/java/appeng/container/implementations/ContainerFluidStorageBus.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2018, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.container.implementations;
+
+
+import java.util.Iterator;
+
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+import appeng.api.AEApi;
+import appeng.api.config.AccessRestriction;
+import appeng.api.config.FuzzyMode;
+import appeng.api.config.SecurityPermissions;
+import appeng.api.config.Settings;
+import appeng.api.config.StorageFilter;
+import appeng.api.config.Upgrades;
+import appeng.api.storage.IMEInventory;
+import appeng.api.storage.channels.IFluidStorageChannel;
+import appeng.api.storage.data.IAEFluidStack;
+import appeng.api.storage.data.IItemList;
+import appeng.container.guisync.GuiSync;
+import appeng.container.slot.OptionalSlotFakeFluid;
+import appeng.container.slot.SlotFakeFluid;
+import appeng.container.slot.SlotRestrictedInput;
+import appeng.parts.misc.PartFluidStorageBus;
+import appeng.util.Platform;
+import appeng.util.helpers.ItemHandlerUtil;
+import appeng.util.iterators.NullIterator;
+
+
+/**
+ * @author BrockWS
+ * @version rv6 - 22/05/2018
+ * @since rv6 22/05/2018
+ */
+public class ContainerFluidStorageBus extends ContainerUpgradeable
+{
+
+	private final PartFluidStorageBus storageBus;
+
+	@GuiSync( 3 )
+	public AccessRestriction rwMode = AccessRestriction.READ_WRITE;
+
+	@GuiSync( 4 )
+	public StorageFilter storageFilter = StorageFilter.EXTRACTABLE_ONLY;
+
+	public ContainerFluidStorageBus( InventoryPlayer ip, PartFluidStorageBus te )
+	{
+		super( ip, te );
+		this.storageBus = te;
+	}
+
+	@Override
+	protected int getHeight()
+	{
+		return 251;
+	}
+
+	@Override
+	protected void setupConfig()
+	{
+		final int xo = 8;
+		final int yo = 23 + 6;
+
+		final IItemHandler config = this.getUpgradeable().getInventoryByName( "config" );

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "config" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)